client/components/phone-input/country-flag.jsx

```diff
@@ -5,68 +5,67 @@
  */
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { Component } from 'react';
 import Gridicon from 'gridicons';
-
 /** Internal Dependencies */
 import Spinner from 'components/spinner';
 
-export default class extends React.Component {
+export default class extends Component {
 	static displayName = 'PhoneInputCountryFlag';
 
 	static propTypes = {
 		countryCode: PropTypes.string.isRequired,
 	};
 
 	state = {
-		ready: false,
-		error: false,
+		flagSvg: null,
 	};
 
-	componentDidUpdate( oldProps ) {
-		if ( this.props.countryCode && this.props.countryCode !== oldProps.countryCode ) {
-			this.setState( { ready: false, error: false } );
-		}
+	componentDidMount() {
+		this.updateFlag();
 	}
 
-	renderSpinner = () => {
-		if ( ( ! this.props.countryCode || ! this.state.ready ) && ! this.state.error ) {
-			return <Spinner size={ 16 } className="phone-input__flag-spinner" />;
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.countryCode !== this.props.countryCode ) {
+			this.updateFlag();
 		}
-	};
-
-	handleImageLoad = () => {
-		this.setState( { ready: true, error: false } );
-	};
+	}
 
-	handleImageError = () => {
-		this.setState( { ready: false, error: true } );
+	updateFlag = () => {
+		this.setState( { flagSvg: null }, () => {
+			// don't confusingly split the import statement with its long string
+			// note that webpack will parse this code at compile-time and auto-generate
+			// a context for loading the SVGs. it will infer the file glob for all SVG
+			// files in the given path and should make a separate bundle for each one
+			//
+			// also, the "noquotes" part of the loader tells it that we're consuming
+			// the SVG data-uri as a JavaScript value and not as a string
+			// see the docs for `svg-url-loader` for more information
+			//
+			// prettier-ignore
+			import(
+				/* webpackChunkName: "svg-flag-[request]" */
```

Comment:
This produces filenames such as `svg-flag-at-svg.js`. Th leading `svg` is kinda redundant, so maybe change to `/* webpackChunkName: "flag-[request]" */`?